modules/turn/turn.c

```diff
@@ -341,22 +341,41 @@ static void tmr_handler(void *arg)
 {
 	struct tmr *tmr = arg;
 	struct le *le;
+	int thrd_id;
 
 	mtx_lock(&turndp()->mutex);
 	if (!turndp()->run)
 		re_cancel();
 
+	thrd_id = GetThreadId(thrd_current());
+
 	/* Reassign one allocation by time */
 	LIST_FOREACH(&turndp()->re_map, le)
 	{
-		struct allocation *al = le->data;
+		struct allocation *al = list_ledata(le);
 		mtx_lock(&al->mutex);
-		udp_thread_attach(al->rel_us);
-		udp_thread_attach(al->rsv_us);
+		udp_thread_attach(al->uks->rel_us);
+		udp_thread_attach(al->uks->rsv_us);
+		al->uks->thrd_id = thrd_id;
+		
 		mtx_unlock(&al->mutex);
 	}
 	list_clear(&turndp()->re_map);
 
+	le = list_head(&turndp()->rm_map);
+	while (le)
+	{
+		struct udp_socks *uks = list_ledata(le);
+		le = le->next;
+		if (thrd_id == uks->thrd_id) {
+			udp_thread_detach(uks->rel_us);
+			udp_thread_detach(uks->rsv_us);
```

Comment:
I think this is a memory leak. instead of detach it should be `mem_deref(uks->rel_us)` and `mem_derf(uks->rsv_us)`